docs/dev/concepts.rst

```diff
@@ -0,0 +1,175 @@
+   client and has access to several pieces of data about the client, such as the
+   data collected by Telemetry_.
+2. A pointer to the :ref:`action <actions>` that this recipe should perform when
+   run.
+3. The configuration data to pass to the Action code when it is executed, called
+   the recipe's **arguments**.
+
+Recipes are stored in the Normandy database and the :ref:`runtime <runtime>`
+fetches them from the service during execution. Here's an example of a recipe in
+JSON format as returned by Normandy:
+
+.. code-block:: json
+
+   {
+      "id": 1,
+      "name": "Console Log Test",
+      "revision_id": 12,
+      "action": {
+         "name": "console-log",
+         "implementation_url": "https://self-repair.mozilla.org/api/v1/action/console-log/implementation/8ee8e7621fc08574f854972ee77be2a5280fb546/",
+         "arguments_schema": {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "description": "Log a message to the console",
+            "type": "object",
+            "properties": {
+               "message": {
+                  "default": "",
+                  "description": "Message to log to the console",
+                  "type": "string",
+               }
+            },
+            "required": [
+               "message"
+            ],
+         }
+      },
+      "arguments": {
+         "message": "It works!"
+      }
+   }
+
+Note that the recipe as returned by the API also contains data about the action,
+including a schema describing the arguments field.
+
+.. _JEXL: https://github.com/TechnologyAdvice/Jexl
+.. _Telemetry: https://wiki.mozilla.org/Telemetry
+
+.. _runtime:
+
+Runtime
+-------
+The runtime is an execution environment that can affect a running instance of
+Firefox. Upon activation (typically a few moments after Firefox launches), the
+runtime:
+
+1. Downloads :ref:`recipes <recipes>` from the Normandy service.
+2. Evaluates the recipe filters and filters out recipes that do not match the
+   client the runtime is installed within.
+3. Downloads the :ref:`actions <actions>` for the remaining recipes.
+4. Executes the action code for each recipe in a sandbox, passing in the
+   arguments from the recipe, and a :doc:`driver object </dev/driver>` containing
+   methods that can perform privileged actions.
+
+The runtime is implemented as a system add-on in the normandy-addon_ repository.
+
+.. _normandy-addon: https://github.com/mozilla/normandy-addon
+
+Threat Model
+------------
+Since the goal of SHIELD is to allow Mozilla to perform certain privileged
+actions quickly without shipping full updates to Firefox, it is a tempting
+target for compromising Firefox users. Normandy includes several security
+controls to help mitigate this risk.
+
+Action and Recipe Signing
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Actions and recipes that are downloaded by the runtime are signed using the
+autograph_ service. The runtime verifies the signature upon downloading the
```

Comment:
I'd refer to the protocol as well `signed according to the Content-Signature protocol as provided by the autograph service`.
Unfortunately, we don't yet have a spec for Content-Signature for you to link to.